src/analysis/convex/strict_convex_space.lean

```diff
@@ -87,74 +87,64 @@ lemma strict_convex_space.of_strict_convex_closed_unit_ball
+    ((convex_closed_ball _ _).to_strict_convex' $ λ x hx y hy hne, _),
+  rw [interior_closed_ball (0 : E) one_ne_zero, closed_ball_diff_ball, mem_sphere_zero_iff_norm]
+    at hx hy,
+  rcases h x y hx hy hne with ⟨a, b, hab, hlt⟩,
+  use b,
+  rwa [affine_map.line_map_apply_module, interior_closed_ball (0 : E) one_ne_zero,
+    mem_ball_zero_iff, sub_eq_iff_eq_add.2 hab.symm]
 end
 
-lemma strict_convex_space.of_norm_add_lt_aux {a b c d : ℝ} (ha : 0 < a) (hab : a + b = 1)
-  (hc : 0 < c) (hd : 0 < d) (hcd : c + d = 1) (hca : c ≤ a) {x y : E} (hy : ∥y∥ ≤ 1)
-  (hxy : ∥a • x + b • y∥ < 1) :
-  ∥c • x + d • y∥ < 1 :=
+lemma strict_convex_space.of_norm_combo_ne_one
+  (h : ∀ x y : E, ∥x∥ = 1 → ∥y∥ = 1 → x ≠ y →
+    ∃ a b : ℝ, 0 ≤ a ∧ 0 ≤ b ∧ a + b = 1 ∧ ∥a • x + b • y∥ ≠ 1) :
+  strict_convex_space ℝ E :=
 begin
-  have hbd : b ≤ d,
-  { refine le_of_add_le_add_left (hab.trans_le _),
-    rw ←hcd,
-    exact add_le_add_right hca _ },
-  have h₁ : 0 < c / a := div_pos hc ha,
-  have h₂ : 0 ≤ d - c / a * b,
-  { rw [sub_nonneg, mul_comm_div, ←le_div_iff' hc],
-    exact div_le_div hd.le hbd hc hca },
-  calc ∥c • x + d • y∥ = ∥(c / a) • (a • x + b • y) + (d - c / a * b) • y∥
-        : by rw [smul_add, ←mul_smul, ←mul_smul, div_mul_cancel _ ha.ne', sub_smul,
-            add_add_sub_cancel]
-    ... ≤ ∥(c / a) • (a • x + b • y)∥ + ∥(d - c / a * b) • y∥ : norm_add_le _ _
-    ... = c / a * ∥a • x + b • y∥ + (d - c / a * b) * ∥y∥
-        : by rw [norm_smul_of_nonneg h₁.le, norm_smul_of_nonneg h₂]
-    ... < c / a * 1 + (d - c / a * b) * 1
-        : add_lt_add_of_lt_of_le (mul_lt_mul_of_pos_left hxy h₁) (mul_le_mul_of_nonneg_left hy h₂)
-    ... = 1 : begin
-      nth_rewrite 0 ←hab,
-      rw [mul_add, div_mul_cancel _ ha.ne', mul_one, add_add_sub_cancel, hcd],
-    end,
+  refine strict_convex_space.of_strict_convex_closed_unit_ball ℝ
+    ((convex_closed_ball _ _).to_strict_convex _),
+  simp only [interior_closed_ball _ one_ne_zero, closed_ball_diff_ball, set.pairwise,
+    frontier_closed_ball _ one_ne_zero, mem_sphere_zero_iff_norm],
+  intros x hx y hy hne,
+  rcases h x y hx hy hne with ⟨a, b, ha, hb, hab, hne'⟩,
+  exact ⟨_, ⟨a, b, ha, hb, hab, rfl⟩, mt mem_sphere_zero_iff_norm.1 hne'⟩
 end
 
-/-- Strict convexity is equivalent to `∥a • x + b • y∥ < 1` for all `x` and `y` of norm at most `1`
-and all strictly positive `a` and `b` such that `a + b = 1`. This shows that we only need to check
-it for fixed `a` and `b`. -/
-lemma strict_convex_space.of_norm_add_lt {a b : ℝ} (ha : 0 < a) (hb : 0 < b) (hab : a + b = 1)
-  (h : ∀ x y : E, ∥x∥ ≤ 1 → ∥y∥ ≤ 1 → x ≠ y → ∥a • x + b • y∥ < 1) :
+lemma strict_convex_space.of_norm_add_ne_two
+  (h : ∀ ⦃x y : E⦄, ∥x∥ = 1 → ∥y∥ = 1 → x ≠ y → ∥x + y∥ ≠ 2) :
+  strict_convex_space ℝ E :=
+begin
+  refine strict_convex_space.of_norm_combo_ne_one
+    (λ x y hx hy hne, ⟨1/2, 1/2, one_half_pos.le, one_half_pos.le, add_halves _, _⟩),
+  contrapose! hne with H,
+  refine not_imp_not.1 (h hx hy) _,
```

Comment:
Sounds like you're contraposing on the wrong thing because you're doing it twice.

Reply:
Fixed.